File: kaop/src/main/java/io/github/coyamo/kaop/Aspect.java
package io.github.coyamo.kaop;


/**
 * @author cxc
 * @date 2022/11/9 17:12
 * @version 1.0
 */
public abstract class Aspect {

    Object advice(AbsJoinPoint point){
        before(point);
        Object result = around (point);
        after(point);
        return result;
    }

    public void before(AbsJoinPoint point){}

    public Object around (AbsJoinPoint point){
        return point.proceed();
    }

    public void after(AbsJoinPoint point){}
}

File: kaop/src/main/java/io/github/coyamo/kaop/Aspect.kt
package io.github.coyamo.kaop


/**
 * 切面
 * @author Coyamo
 * @date 2022/12/17 00:47
 * @version 1.0
 */
abstract class Aspect {
    /**
     * 开始处理
     */
    fun advice(point: AbsJoinPoint): Any? {
        before(point)
        val result = around(point)
        after(point)
        return result
    }

    open fun before(point: AbsJoinPoint) {}

    open fun around(point: AbsJoinPoint): Any? {
        return point.proceed()
    }

    open fun after(point: AbsJoinPoint) {}
}